### PR TITLE
fix: chown shenanigans

### PR DESCRIPTION
**Pull request by hydazz**
spawns chown in the background as a 'lazy' way of hopfully negating init stalls
also format scripts and update arm dockerfile

### CODE REVIEW

**Comment by hydazz** [comment]
also fixing the startup order, should go:
init-check-variables > init-config-immich > init-test-run > init-video-immich > services

**Comment by hydazz** [comment]
https://github.com/imagegenius/docker-immich/issues/394

**Review by martabal — APPROVED** [review]
Nice

**Comment by hydazz** [comment]
@martabal you think running chown in the background is the best way?

**Comment by martabal** [comment]
> @martabal you think running chown in the background is the best way?

IMO, it is not. Someone who misconfigured his share won't realize this issue and this solution will impact people who don't have permission issues.

**Comment by hydazz** [comment]
@martabal how about dat

**Comment by hydazz** [comment]
How about this*

**Comment by martabal** [comment]
I think we'll have the same issue #273. We probably want to check if `/photos` have the correct permissions before changing them

**Comment by hydazz** [comment]
this is not a recursive chown, its just the photos folder, and subdirectories 1 depth

**Comment by martabal** [comment]
> this is not a recursive chown, its just the photos folder, and subdirectories 1 depth

Will this work for files not owned by the `abc` user (for exemple files in a nfs mounted by root but with Immich started with a different user)?

**Comment by hydazz** [comment]
depends on how the nfs permissions are setup, the init scripts are ran as root, so if root has permission to modify the nfs files, then yes, its not being done by abc
```
cd Immich/
bash: cd: Immich/: Permission denied

whoami
root
```

```
cd Immich/

pwd
/mnt/nfs/Picard/Media/Immich

whoami
sandy
```

long story short, the NFS ACL need to match the UID of the local user set in the `PUID`

Screenshot 2024-07-12 at 6 14 13 pm

chown is not the way to edit/fix permissions on NFS shares, but we cant check if the mount is nfs soo...

**Review by martabal — APPROVED** [review]

**Review by martabal — CHANGES_REQUESTED** [review]

File: root/etc/s6-overlay/s6-rc.d/svc-microservices/run [hunk 1]
```diff
@@ -4,10 +4,10 @@
 lib_path="/usr/lib/$(arch)-linux-gnu/libmimalloc.so.2"
 export LD_PRELOAD="$lib_path"
 
-export CPU_CORES="${CPU_CORES:=$(/app/immich/server/scripts/get-cpus.sh)}"
-echo "Detected CPU Cores: $CPU_CORES"
-if [ "$CPU_CORES" -gt 4 ]; then
-  export UV_THREADPOOL_SIZE=$CPU_CORES
+export "${CPU_CORES:="$(/app/immich/server/scripts/get-cpus.sh)"}"
```
[the remark this answers is not in the document]
Comment (hydazz) [reply root/etc/s6-overlay/s6-rc.d/svc-microservices/run]: i assume the `:` should not be there?

**Review by martabal — CHANGES_REQUESTED** [review]
It still does not work `./run: line 7: 8: command not found`

**Comment by hydazz** [comment]
> It still does not work `./run: line 7: 8: command not found`

my bad

**Comment by hydazz** [comment]
any other changes or good to merge?

**Comment by martabal** [comment]
Another issue caused during the first install: `/usr/bin/find: ‘/photos/*’: No such file or directory`

File: root/etc/s6-overlay/s6-rc.d/init-config-immich/run [hunk 2]
```diff
@@ -4,10 +4,10 @@
...
     /config/machine-learning/models \
-    "$IMMICH_MEDIA_LOCATION"
+    "${IMMICH_MEDIA_LOCATION}"
 
 # permissions
-find "$IMMICH_MEDIA_LOCATION" -maxdepth 0 \( ! -user abc -o ! -group abc \) -exec lsiown -R abc:abc {} \;
 find /app/immich -path "*/node_modules" -prune -o -exec chown abc:abc {} +
 lsiown -R abc:abc \
     /config
+[[ "${APPLY_PERMISSIONS}" == "true" ]] && lsiown -R abc:abc "${IMMICH_MEDIA_LOCATION}"
```
Comment (martabal) [inline root/etc/s6-overlay/s6-rc.d/init-config-immich/run:13]: I'm fine with it but maybe we should have it for `/config` too ?

**Review by martabal — APPROVED** [review]

File: root/etc/s6-overlay/s6-rc.d/init-config-immich/run (hunk 2 above)
Comment (hydazz) [reply root/etc/s6-overlay/s6-rc.d/init-config-immich/run:13]: > 

/config should always be mounted locally, and not our problem if mounted any other way IMO

**Pull request merged**